src/lua/utils.c

```diff
@@ -285,6 +286,19 @@ luaL_setcdatagc(struct lua_State *L, int idx)
 	lua_pop(L, 1);
 }
 
+int
+luaT_setfuncs(struct lua_State *L, const struct luaL_Reg *funcs)
```

Comment:
It looks like `luaL_setfuncs` with `nup` being 0, doesn't it?

Reply:
Side note: Another one feature from 5.2:)

Reply:
Yes, it is quite similar to `luaL_setfuncs()` from Lua 5.2 (and from LuaJIT). I'll explain why I added my own function and we'll make a quorum decision :)

I want a helper function, which is easy to understand, when you look at code that uses it[^1]. It should have a description and a simple source, so I can go to declaration/definition and spot in seconds what exactly it is doing. The LuaJIT's `luaL_setfuncs()` has neither of those properties: it has no description in the source code and it has a bit non-obvious source due to the `nup` argument handling.

(I would also note that it is not very clear that one should find the documentation in the Lua 5.2 manual -- if we'll ignore the fact that this description is out of the sources and you can't go to it in a click.)

So, for my taste, presence of a technically satisfying function in LuaJIT doesn't mean that using it makes my code better comparing to using a simple and well documented own helper.

Dixi.

[^1]: Yes, I got tired by deducing `luaL_register()`'s behavior from its source code.

Reply:
I totally agree with you regarding `luaL_register`, so I said no word regarding `luaT_newmodule`. However, I still can't agree with introducing `luaT_setfuncs`. As you mentioned, the latter function has quite the same source code and (IMHO) handling `nup` parameter makes it as much intricate as `lua_pushcclosure` usage is more complex than `lua_pushcfunction`.

Generally, I'm OK with introducing helpers, that provides similar functionality but results to quite a different effect (`luaL_register` and `luaL_register_module` are just a hell), but introducing the function with 2 parameters instead of using the one provided by LuaJIT C API with 3 parameters looks overkill to me..

Reply:
I'll try to suggest the solution, that may help all here to stay satisfied...
Can we introduce our own helper `luaT_setfuncs(L, L_Reg)` which simply calls `luaL_setfuncs(L, L_Reg, 0)`?
Here, we can add the comment inside our helper, where we can take a look to the documentation and the description of the last parameter.

Reply:
@Buristan, this is not a compromise at all. I'm for using LuaJIT interfaces in Tarantool; @Totktonada wants to use the functions that have the clear signature and are easy to understand.

Your proposal contradicts both points :)

Reply:
I'm motivated by two points:

* Documentation.
* Simple source code.

If I'll use `luaL_setfuncs()` in `luaT_setfuncs()`, the latter point would not be in effect IMHO. And introducing the helper just because of one of those points looks not motivated enough.

Okay. Let's go with `luaL_setfuncs(<...>, 0)`. It is not so hard to understand as `luaL_register()` and I can agree.

Sooner or later we should do something to improve in-source LuaJIT's API documentation.

Reply:
I've replaced `luaT_setfuncs()` with `luaL_setfuncs()` and updated commits messages (mainly the first commit).

Reply:
Seems resolved now. Closing the discussion.